script.js

```diff
@@ -358,3 +358,68 @@ Object.keys(navLinks).forEach((k) => {
     navLinks[k].addEventListener('click', closeMobileMenu);
   }
 });
+
+// FORM VALIDATION & LOCAL STORAGE
+const formOne = document.getElementById('formOne');
+const inputName = document.getElementById('name');
+const email = document.getElementById('email');
+const textArea = document.getElementById('textArea');
+const lStorage = window.localStorage.getItem('data');
+const formEntry = {};
+let errorMsg = document.getElementById('errorMsg');
+let emailTest = '';
+let formTest = false;
+
+// eslint-disable-next-line no-trailing-spaces
+/* Get lowercase of each vale in 
+email for testing real user case value */
+
+// Validation Starts
+email.addEventListener('keyup', () => {
+  emailTest = email.value.toLowerCase();
+});
+
+formOne.addEventListener('submit', (e) => {
+  if (emailTest !== email.value) {
+    errorMsg.innerText = 'Message goes here';
+    if (formTest === false) {
+      errorMsg.classList.toggle('hideErr');
+    }
+    formTest = true;
+    // To remove the errMsg when email input is clicked
+    email.addEventListener('click', () => {
+      if (formTest === true) {
+        errorMsg.classList.toggle('hideErr');
+      }
+      formTest = false;
+    });
+    e.preventDefault();
+  } else errorMsg = '';
+});
+
+// Local Storage Get user input Data
+function getUserData() {
+  formEntry[inputName.name] = inputName.value;
+  formEntry[email.name] = email.value;
+  formEntry[textArea.name] = textArea.value;
+  let items = JSON.stringify(formEntry);
+  localStorage.setItem('data', `${items}`);
+}
```

Comment:
- [ ] You have done a great job storing all the user input data in a single local storage registry, however the form data for all inputs should be stored in a single object too eg: 
```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```
[HINT] You can simply do it by setting key value pairs, with empty values:
```
const formData = {
    name: "",
    email: "",
    body: "",
}; 
```
then assign user inputs to object keys as values, then update the local storage: 
`localStorage.setItem('formData', JSON.stringify(formData));`

Reply:
I saved the user input inside a formData with is an object I called formEntry. i initialized an empty object on line 368 as formEntry = {} . so user input are stored in an object before I stringified the collected formEntry object into the localStorage. Please give me more clarification because I feel you want me to store user data inside an object of with I did. please make clarify your observation.

Reply:
and make use of keys to input them into the object. making use of the input name as specified in my HTML as the key and the input value as the value. that is accessing the name property and also the value property of the input and textarea in the form

Reply:
I have checked, sorry I missed that part, you are adding an object after parsing. Please open a new review request and it will be approved. Sorry for making you confused )

Reply:
no problem Shoira thanks for understanding and your patience. thank you

Reply:
Sorry, we missed your point earlier. Your understanding is amazing. Keep that up:+1: